api/outdated/outdated/models.py

```diff
@@ -10,57 +14,141 @@
 
     class Meta:
         ordering = ["name", "id"]
+        unique_together = ("name", "provider")
 
     def __str__(self):
         return self.name
 
 
-class DependencyVersion(UUIDModel):
-
+class ReleaseVersion(UUIDModel):
     dependency = models.ForeignKey(Dependency, on_delete=models.CASCADE)
-    version = models.CharField(max_length=100)
-    release_date = models.DateField()
-    end_of_life_date = models.DateField()
+    major_version = models.IntegerField()
+    minor_version = models.IntegerField()
+    _latest_patch_version = models.IntegerField(editable=False, null=True, blank=True)
+    last_checked = models.DateTimeField(
+        editable=False,
+        null=True,
+        blank=True,
+        default=get_yesterday,
+    )
+
     status = models.CharField(max_length=10, choices=STATUS_CHOICES, editable=False)
+    end_of_life = models.DateField(null=True, blank=True)
+
+    def __str__(self) -> str:
+        return f"{self.dependency.name} {self.version}"
+
+    @property
+    def version(self):
+        return f"{self.major_version}.{self.minor_version}"
+
+    @property
+    def latest_patch_version(self):  # pragma: todo cover
+        if self.last_checked < timezone.now() - timedelta(days=1):
+            self.last_checked = timezone.now()
+            self._latest_patch_version = get_latest_patch_version(self)
+            self.save()
+        return self._latest_patch_version
 
     class Meta:
-        ordering = ["end_of_life_date", "dependency__name", "version", "release_date"]
-        unique_together = ("dependency", "version")
+        ordering = [
+            "end_of_life",
+            "dependency__name",
+            "major_version",
+            "minor_version",
+        ]
+        unique_together = ("dependency", "major_version", "minor_version")
 
     @property
     def status(self):
-        if date.today() >= self.end_of_life_date:
+        if not self.end_of_life:
+            return STATUS_OPTIONS["undefined"]
+        elif date.today() >= self.end_of_life:
             return STATUS_OPTIONS["outdated"]
-        elif date.today() + timedelta(days=30) >= self.end_of_life_date:
+        elif date.today() + timedelta(days=150) >= self.end_of_life:
             return STATUS_OPTIONS["warning"]
         return STATUS_OPTIONS["up_to_date"]
 
+
+class Version(UUIDModel):
+    release_version = models.ForeignKey(ReleaseVersion, on_delete=models.CASCADE)
+    patch_version = models.IntegerField()
+    release_date = models.DateField(null=True, blank=True)
+
+    class Meta:
+        ordering = [
+            "release_version__end_of_life",
+            "release_version__dependency__name",
+            "release_version__major_version",
+            "release_version__minor_version",
+            "patch_version",
+        ]
+
     def __str__(self):
-        return self.dependency.name + self.version
+        return f"{self.release_version.dependency.name} {self.release_version.version}.{self.patch_version}"
```

Comment:
```suggestion
        return f"{str(self.release_version)}.{self.patch_version}"
```

Reply:
Done in https://github.com/adfinis/Outdated/pull/21/commits/93eb01ba179cbd96e612621064cb0f9cfa4e10d2.

The str() is not necessary, as f-strings use the string representation of classes by default.